### PR TITLE
Enabled DI into properties from YAML

### PR DESCRIPTION
**Pull request by FrankBakkerNl**
## Proposed change

Enable DI for properties loaded from yaml. 
Specifically needed so that V3 Entities can be configured and receive a IHaContext from the container

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/DevelopmentApps/apps/CodeGen.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using JoySoftware.HomeAssistant.Client;
+using NetDaemon.Common;
+
+namespace NetDaemon.DevelopmentApps.apps
+{
+    [NetDaemonApp]
+    public class CodeGen : IAsyncInitializable
```
Comment (Zorgino) [inline src/DevelopmentApps/apps/CodeGen.cs:8]: it's presence in this PR intended?

File: src/Daemon/NetDaemon.Daemon/Daemon/Config/YamlAppConfig.cs [hunk 2]
```diff
@@ -81,13 +81,15 @@ public IEnumerable<ApplicationContext> GetInstances()
...
             var appContext = _appInstantiator.Instantiate(netDaemonAppType, appId);
 
-            SetPropertyConfig(appNode, appId, appContext.ApplicationInstance);
+            SetPropertyConfig(appNode, appId, appContext);
             
             return appContext;
         }
 
-        public void SetPropertyConfig(YamlMappingNode appNode, string? appId, object appInstance)
+        public void SetPropertyConfig(YamlMappingNode appNode, string? appId, ApplicationContext applicationContext)
```
Comment (Zorgino) [inline src/Daemon/NetDaemon.Daemon/Daemon/Config/YamlAppConfig.cs:89]: appId can be obtained from applicationContext

File: src/DevelopmentApps/apps/CodeGen.cs (hunk 1 above)
Comment (FrankBakkerNl) [reply src/DevelopmentApps/apps/CodeGen.cs:8]: Nope

File: src/App/NetDaemon.App/Common/ApplicationContext.cs [hunk 3]
```diff
@@ -91,25 +118,18 @@ public bool IsEnabled
...
-            {
-                trackedDisposable.Dispose();
-            }
         }
 
+        /// <inheritdoc />
         public void Dispose()
         {
-            DisposeAsync().AsTask().Wait();
+            _serviceScope.Dispose();
```
Comment (BeeHiveJava) [inline src/App/NetDaemon.App/Common/ApplicationContext.cs:132]: Should we dispose the `ApplicationInstance` here as well?

File: src/Daemon/NetDaemon.Daemon/Daemon/NetDaemonHost.cs [hunk 4]
```diff
@@ -119,15 +120,24 @@ internal async Task WaitForTasksAsync()
...
+            var applicationContext = ApplicationContext.CreateFromAppInstance(app, ServiceProvider);
             InternalRunningAppInstances[applicationContext.Id!] = applicationContext;
             InternalAllAppInstances[applicationContext.Id!] = applicationContext;
         }
 
+        internal T LoadApp<T>(string id)
+        {
+            var applicationContext = new ApplicationContext(typeof(T), id, ServiceProvider);
+            InternalRunningAppInstances[applicationContext.Id!] = applicationContext;
+            InternalAllAppInstances[applicationContext.Id!] = applicationContext;
```
Comment (BeeHiveJava) [inline src/Daemon/NetDaemon.Daemon/Daemon/NetDaemonHost.cs:136]: What do we want to happen here if an app already exists? Or is that not a realistic scenario to expect?

Comment (FrankBakkerNl) [reply src/Daemon/NetDaemon.Daemon/Daemon/NetDaemonHost.cs]: This method is also just for unit testing ( I will update the comment). We could probably find a better solution for this eventually

File: src/App/NetDaemon.App/Common/ApplicationContext.cs (hunk 3 above)
Comment (FrankBakkerNl) [reply src/App/NetDaemon.App/Common/ApplicationContext.cs]: Good point, indeed it should

**Pull request merged**